### PR TITLE
A minor refactor of `CloneTmaCircularBufferLoopAndInsertSync`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
I am trying to avoid directly referring `loop_type_` as much as possible. That is, instead of thinking like:

> because this is the main loop, I need to create a mbarrier arrive after the last read for WAR

I prefer to think like:

> because this loop is involved in WAR, and it has a compute, I need to insert a mbarrier arrive after the compute for WAR

This way, we can easily extend `CloneTmaCircularBufferLoopAndInsertSync` when we add more circular buffer loop stages, while maintaining code readability.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!test

**Comment by zasdfgbnm** [comment]
!test

**Comment by zasdfgbnm** [comment]
!test

**Comment by zasdfgbnm** [comment]
!test

**Comment by zasdfgbnm** [comment]
!test

File: csrc/device_lower/pass/circular_buffer.cpp [hunk 1]
```diff
@@ -334,6 +334,26 @@ class CloneTmaCircularBufferLoopAndInsertSync
         war_mbarriers_to_uses_(getAllWarMbarriersToUses()),
         war_mbarriers_to_wait_(getAllMbarriersToWait()) {}
 
+  bool clonesCircularBufferLoad() const {
+    return loop_type_ == CircularBufferLoopStage::Prolog ||
+        loop_type_ == CircularBufferLoopStage::Main;
+  }
+
+  bool clonesCompute() const {
```
Comment (rdspring1) [inline csrc/device_lower/pass/circular_buffer.cpp:342]:
```suggestion
  // The compute expressions are cloned for these circular buffer loop types.
  // e.g., No actual computation occurs in the Prologue stage.
  bool clonesCompute() const {
```

File: csrc/device_lower/pass/circular_buffer.cpp [hunk 2]
```diff
@@ -334,6 +334,26 @@ class CloneTmaCircularBufferLoopAndInsertSync
         war_mbarriers_to_uses_(getAllWarMbarriersToUses()),
         war_mbarriers_to_wait_(getAllMbarriersToWait()) {}
 
+  bool clonesCircularBufferLoad() const {
```
Comment (rdspring1) [inline csrc/device_lower/pass/circular_buffer.cpp:337]:
```suggestion
  // The circular buffer load expressions are cloned for these circular buffer loop types.
  // e.g., No additional loads are required for the Epilogue stage.
  bool clonesCircularBufferLoad() const {
```

**Review by rdspring1 — APPROVED** [review]
🎉

File: csrc/device_lower/pass/circular_buffer.cpp (hunk 1 above)
Comment (zasdfgbnm) [reply csrc/device_lower/pass/circular_buffer.cpp]: I moved the logic to `type.h` and added comments there.

**Comment by zasdfgbnm** [comment]
!test

**Comment by zasdfgbnm** [comment]
!test

**Comment by zasdfgbnm** [comment]
!test

**Pull request merged**